build/Microsoft.DotNet.Cli.BuildInfo.targets

```diff
@@ -0,0 +1,40 @@
+<Project ToolsVersion="15.0">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.tasks))\dir.tasks" />
+
+  <Target Name="WriteBuildInfoProps"
+          Condition=" !Exists('$(BuildInfoProps)') "
```

Comment:
we need to rework this condition. CommitCount and CommitHash can change from build-to-build. I think we should always run this target, should always get the current CommitCount and CommitHash, but only write hte props file IF it does not exist OR it has inconsistent values. This means that we need to conditionally import the props file into this targets file so we can inspect the values.